src/librustc_mir/borrow_check/nll/region_infer/values.rs

```diff
@@ -215,11 +214,7 @@ impl RegionValues {
                 RegionVid::new(num_region_variables),
                 RegionElementIndex::new(elements.num_elements()),
             ),
-            causes: if track_causes.0 {
-                Some(CauseMap::default())
-            } else {
-                None
-            },
+            causes: Some(CauseMap::default()),
```

Comment:
This seems ok but mildly surprising -- that is, I think we should adjust some comments. I guess I would like a comment on `new` like this:

---

Creates a new set of "region values" that tracks causal information. Each of the regions in `num_region_variables` will be initialized with an empty set of points and no causal information.

---

Then I would expect to remove the `Clone` impl and add a new method like:

```
// This is what we use at first...
fn clone_not_tracking_causes(&self) -> Self { .. }

// ...this is what `clone` is today
fn clone_tracking_causes(&self) -> Self { .. }
```